cmake/cmake_options.yml

```diff
@@ -26,7 +26,7 @@ USE_SSE: "ON" # Build with x86 SSE instruction support IF NOT ARM
 USE_F16C: "ON" # Build with x86 F16C instruction support) # autodetects support if "ON"
 USE_LAPACK: "ON" # Build with lapack support
 USE_MKL_IF_AVAILABLE: "ON" # Use MKL if found
-USE_MKLML_MKL: "ON" # Use MKLDNN variant of MKL (if MKL found) IF USE_MKL_IF_AVAILABLE AND (NOT APPLE)
+USE_MKLML: "ON" # Use MKLDNN variant of MKL (if MKL found) IF USE_MKL_IF_AVAILABLE AND (NOT APPLE)
```

Comment:
Not sure about that one, especially considering it breaks compatibility and hides the initial intention to only use the mkl version
@TaoLv

Reply:
There are no guarantees currently about CMake options compatibility.

> hides the initial intention to only use the mkl version
I'm not sure I understand, what intention do you mean?

Reply:
The comment states that this option is to specifically use the mkldnn variant of mkl and not the standalone mkldnn. 

You are right that we do not guarantee compatibility, but I don't see a good reason here why we should break it intentionally. Could you elaborate?

Reply:
Agree that compatibility is more important, although `USE_MKLML_MKL` is really redundant for me. Is it possible to give a warning message if `USE_MKLML_MKL` is provided and set its value to USE_MKLML? And remove this warning message and `USE_MKLML_MKL` totally in a major release?

Reply:
> The comment states that this option is to specifically use the mkldnn variant of mkl and not the standalone mkldnn.
> 
> You are right that we do not guarantee compatibility, but I don't see a good reason here why we should break it intentionally. Could you elaborate?

Because the name does not make sense and leads to confusion.

> The comment states that this option is to specifically use the mkldnn variant of mkl and not the standalone mkldnn.

This is an example of the confusion. There is no such thing as *mkldnn variant of mkl*. There is MKLDNN - an independent addition to MKL. And there is a packed (and easier to download) version of MKL - MKLML.

What is and was wrong though is the comment about this variable, since the option is not related to MKLDNN.

Reply:
> Is it possible to give a warning message if USE_MKLML_MKL is provided and set its value to USE_MKLML?

I have added a warning.